lua/nvim-treesitter/parsers.lua

```diff
@@ -707,6 +707,7 @@ local M = {
 }
 
 function M.ft_to_lang(ft)
+  ft = vim.split(ft, '.', true)[1]
```

Comment:
We could use a comment here that vim's filetypes can be `.` separated. And this looks like a case where we may have different filetypes per-buffer :thinking:

Reply:
I don't think we really care about the part after the `.` tbh.

My reasoning here is that vim will try to guesstimate what the "inner" languages are based on regexes, which we _know_ are less accurate than what we can do with tree-sitter. Thus we don't care about what is after this first part of the filetype.

Reply:
Thinking more about this, the multi lang thing expects the syntax to be combined, but with treesitter it doesn't make sense to combine things, but to use a different parser that inherits another one.

For the example in the issue it doesn't break, but for the other example it does https://github.com/nvim-treesitter/nvim-treesitter/pull/1771#issuecomment-910360775. I think explicit is better here, and ruby.spect should be added to the filetype list of parsers.lua:ruby (if the grammar is really the same as ruby).

Reply:
I am not really familiar with how treesitter works so don't have much insight on this part ... but I know from a snippets standpoint it does utilize inheritance.  That is, my `ruby` snippets are included when I am in a `ruby.rspec` filetype _in addition_ to snippets specific to `rspec` (that are not present in a `ft=ruby` file).

In terms of highlighting, I could see a similar benefit regardless of how it is achieved.

Reply:
For plugins this is great, but for tree-sitter hightlight I think it is better to have a proper parser for different filetypes instead of having two parsers hightlighting the same file.

If the `ruby.rspec` works with the ruby grammar without errors, you can just add `used_by = { "ruby.rspec" }` here

https://github.com/nvim-treesitter/nvim-treesitter/blob/06e84da7d7c0302bb89ac39f8c3a21d5c557ab5b/lua/nvim-treesitter/parsers.lua#L164-L170

Reply:
I don't really like doing it this way because of the lack of genericity.
The vim doc clearly states that dot-separated filetypes are a standard, so I don't see why we could not use that.

Reply:
Regarding the inheritance, it would mean that `cpp.doxygen`, `java.doxygen` would mean that the doxygen regex highlights are added in addition to the language highlights. Maybe this could be useful in future to conditionally inject some document parsers.